src/ruuvitag_ble/parser.py

```diff
@@ -59,9 +72,11 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
             native_unit_of_measurement=Units.ELECTRIC_POTENTIAL_MILLIVOLT,
             native_value=decoder.battery_voltage_mv,
         )
-        self.update_sensor(
-            key="movement_counter",
-            device_class=DeviceClass.COUNT,
-            native_unit_of_measurement=None,
-            native_value=decoder.movement_counter,
-        )
+
+        if isinstance(decoder, DataFormat5Decoder):
```

Comment:
This shouldn't depend on the decoder class version, but whether it knows about movement counting.
```suggestion
        if hasattr(decoder, "movement_counter"):
```